tests/test-umockdev-record.vala

```diff
@@ -202,11 +202,17 @@ t_system_single ()
     assert_in("E: DEVNAME=/dev/zero", sout);
 #if HAVE_SELINUX
     // we may run on a system without SELinux
+    int res = -1;
     if (FileUtils.test("/sys/fs/selinux", FileTest.EXISTS)) {
         string context;
-        assert_cmpint (Selinux.lgetfilecon ("/dev/null", out context), CompareOperator.GT, 0);
-        assert_in("E: __DEVCONTEXT=" + context + "\n", sout);
-    } else {
+        res = Selinux.lgetfilecon ("/dev/null", out context);
+
+        if (res > 0) {
+            assert_in("E: __DEVCONTEXT=" + context + "\n", sout);
+        }
```

Comment:
Does it fail with something useful, like `ENOSYS` or so? that could be asserted in the failing case.

Reply:
Yes, it fails with `ENODATA`. Would you prefer adding it here explicitly (and fail if errno is not `ENODATA`/`ENOTSUP`)? I think it might be enough to catch these corner cases :smile:

Reply:
... and check that the error is one of these, i.e. we know that it's due to "no selinux", not due to some other stupid stuff like ENOENT. After all, all of these devices are emulated, so I'd like to make sure that failures are under tight control.